src/lib.rs

```diff
@@ -966,3 +1028,18 @@ annotations = { file = "Runefile.yml", kind = "yaml" }
         );
     }
 }
+
+#[test]
+fn test_package_name_parse() {
```

Comment:
Let's add more test cases based on [the package naming rules doc](https://www.notion.so/wasmer/Naming-and-Ownership-Policy-ec6e449d8d5f4acd9a616ffa73612f5d).

Success:

- `namespace/package`
- `name-space/pack-age`
- `name_space/pack_age`
- `_/package`

Errors:
- `namespace/package/something-else` (only 2 segments)
- `name space/pack age` (i.e. no spaces)
- `namespace/`
- `/package`
- `namespace/_` (the name can't be `_`)
- `_namespace/_package` (can't have leading/trailing `_` or `-`)
- `namespace_/package_`
- `-namespace/-package`
- `namespace-/package-`
- `http://namespace`
- `nàméspàcë/pàckàgê` (i.e. only `-`, `_`, and ascii alphanumeric characters)
- (the empty string)
- `abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijz/abcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijabcdefghijz` (i.e. namespace and packages are 101 characters when they can be at most 100 characters)

Reply:
ok done